dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/WebContainer.java

```diff
@@ -808,18 +812,43 @@ public void handleRequest(IRequest req, IResponse res, VirtualHost vhost, Reques
             // Begin 293696 ServletRequest.getPathInfo() fails WASCC.web.webcontainer
             String reqURI = req.getRequestURI();
             String decodedReqURI = null;
+           
             if (WCCustomProperties.DECODE_URL_PLUS_SIGN) {
                 decodedReqURI = URLDecoder.decode(reqURI, encoding);
             } else {
                 decodedReqURI = WSURLDecoder.decode(reqURI, encoding);
             }
 
+            //Servlet 6.0 - process after decoded uri
+            if (isServlet60orAbove) {
+                if (isTraceOn && logger.isLoggable(Level.FINE)) { 
+                    logger.logp(Level.FINE, CLASS_NAME, "handleRequest", "decoded uri [" + decodedReqURI + "]" );
+                }
+
+                String path;
+                try {
+                    path = canonicalizeURI(decodedReqURI);
+                }
+                catch (IOException ioe) {
+                    logger.logp(Level.INFO, CLASS_NAME, "handleRequest", "canonicalize sending 400 [" + ioe.getMessage() + "]");
```

Comment:
If we're logging an INFO message to the logs it should be translated.

Reply:
change to  Level.FINE